src/docker/Dockerfile

```diff
@@ -52,43 +52,31 @@ RUN apt-get update && apt-get install -y \
     automake autoconf
 
 # Git 2.0
-RUN apt-get install -y software-properties-common python-software-properties
+RUN apt-get update && apt-get install -y \
+    software-properties-common python-software-properties
 RUN add-apt-repository ppa:git-core/ppa
 RUN apt-get update && apt-get install -y git git-core
 
-# SSH Daemon
-RUN apt-get install -y ssh && \
-  mkdir /var/run/sshd && \
-  chmod 0755 /var/run/sshd
+# Java
+RUN apt-get update && apt-get install -y \
+    openjdk-7-jdk="${OPENJDK_VERSION}"
 
-# Ruby
-RUN apt-get install -y libmagic-dev zlib1g-dev
+# Ruby; for puzzle creation
+RUN apt-get update && apt-get install -y \
+    libmagic-dev zlib1g-dev
 RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get install -y ruby2.2 ruby2.2-dev
-RUN gem update
+RUN apt-get update && apt-get install -y \
+    ruby2.2 ruby2.2-dev
 RUN gem install nokogiri
 RUN gem install bundler
 
-# Java
-RUN apt-get install -y default-jdk
-RUN apt-get install -y gdebi && \
+# Maven
+RUN apt-get update && apt-get install -y gdebi && \
   wget --quiet http://ppa.launchpad.net/natecarlson/maven3/ubuntu/pool/main/m/maven3/maven3_3.2.1-0~ppa1_all.deb && \
   gdebi --non-interactive maven3_3.2.1-0~ppa1_all.deb && \
   ln -s /usr/share/maven3/bin/mvn /usr/bin/mvn && \
   rm -rf maven3_3.2.1-0~ppa1_all.deb
 
-# PhantomJS
-RUN apt-get install -y phantomjs
-
-# S3cmd for AWS S3 integration
-RUN apt-get install -y s3cmd
```

Comment:
@amihaiemil it's needed here: https://github.com/jcabi/jcabi-aether/blob/master/.rultor.yml#L38